### PR TITLE
`General`: Fix broken refresh button in course details

### PR DESCRIPTION
**Pull request by pal03377**
#### General

- [X] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [X] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [X] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [X] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).

### Motivation and Context

@krusche pointed out an issue with the refresh button on course pages: Pressing the refresh button didn't trigger a request to the server and just caused the loading spinner to spin forever.

### Description

My investigation showed that the issue was in #6318, which changed the way courses are loaded on the client: 
Previously, the request to `findOneForDashboard` was directly consumed, because it was `.subscribe`d to. The change was to use a `.pipe` instead for simpler error handling. This however also meant that the actual course result would only be fetched after the `loadCourse` result was consumed = subscribed to. This was not the case for the refresh function.

This PR adds an artificial `subscribe` to the generated observable to change the behavior back again and fix the course refresh. It also adds a test so that this issue won't happen in the future anymore.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Instructor
- 1 Course

1. Log in to Artemis
2. Navigate to the course
3. Click "refresh". Verify that the course is refreshed and not loading forever.
4. Modify the course in some way as the instructor (e.g. change the title).
5. Verify that the refresh triggers the change.
6. Go to #6318 and verify that the testing steps there are unaffected.

#### Performance Review
- [X] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 

#### Code Review
- [X] Code Review 1
- [X] Code Review 2
#### Manual Tests
- [X] Test 1
- [X] Test 2

### Test Coverage

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| course-overview.component.ts | 83.57% | ✅              |

**Note: The coverage is too low, but it will be improved in #6491 to be > 90%.**

### Screenshots

After clicking "refresh":

Before, forever:
image

Now, after some short waiting time:
screenshot-2023-04-26_001261
(refresh finished)

### CODE REVIEW

**Review by YeetTheFirst21 — DISMISSED** [review]
Tested on TS2, also checked all test cases on the previous pr and all seems to be working as expected. See the screenshots attached below for the given errors and refresh working through network tab and visually.

Screenshot 2023-04-26 224309
Screenshot 2023-04-26 224342
Screenshot 2023-04-26 224735
Screenshot 2023-04-26 224812
Screenshot 2023-04-26 224907
Screenshot 2023-04-26 224921

**Comment by pal03377** [comment]
> Step 5 never happens:
> 5. Note that a confirmation dialog opens up and that dialog includes the registration confirmation message from the course settings.

Thanks for testing! This is intended behavior, as [there was no confirmation message given in the settings of that course](https://artemis-test2.artemis.in.tum.de/course-management/8695). I just tested adding one myself again and if a confirmation message is present, the dialog is indeed shown.

**Review by FelixTJDietrich — APPROVED** [review]
The previous issue I commented seems to be intended behaviour. Approved.

File: src/main/webapp/app/overview/course-overview.component.ts [hunk 1]
```diff
@@ -249,6 +249,10 @@ export class CourseOverviewComponent implements OnInit, OnDestroy, AfterViewInit
                 return throwError(() => error);
             }),
         );
+        // Start fetching, even if we don't subscribe to the result.
+        // This enables just calling this method to refresh the course, without subscribing to it:
+        observable.subscribe();
```
[the remark this answers is not in the document]
Comment (pal03377) [reply src/main/webapp/app/overview/course-overview.component.ts]: Good idea, thanks! I implemented it in [this commit](https://github.com/ls1intum/Artemis/pull/6490/commits/f799469abfc189c9cfbeccd1c27c393a79fde9f6).

**Review by sven0311 — APPROVED** [review]
Code looks good and works like expected

**Review by lennart-keller — APPROVED** [review]

**Review by FelixTJDietrich — DISMISSED** [review]
Still works as intended on ts2. Reapproved.

**Review by dearjasmina — DISMISSED** [review]
works as expected, tested on server 2

File: src/main/webapp/app/overview/course-overview.component.ts [hunk 2]
```diff
@@ -249,6 +251,10 @@ export class CourseOverviewComponent implements OnInit, OnDestroy, AfterViewInit
                 return throwError(() => error);
             }),
         );
+        // Start fetching, even if we don't subscribe to the result.
+        // This enables just calling this method to refresh the course, without subscribing to it:
+        this.loadCourseSubscription = observable.subscribe();
```
Comment (bassner) [inline src/main/webapp/app/overview/course-overview.component.ts:256]: If you unsubscribe in the `ngOnDestroy`, we should also check if `loadCourseSubscription` already contains a subscription from a previous refresh round, and unsubscribe from it. What do you think?

```suggestion
        this.loadCourseSubscription?.unsubscribe();
        this.loadCourseSubscription = observable.subscribe();
```

**Review by bassner — CHANGES_REQUESTED** [review]

**Review by FelixTJDietrich — APPROVED** [review]
Changes look good to me and it still works as expected. Re-approved.

**Review by JohannesStoehr — APPROVED** [review]
Code looks good to me.
My comment about unsubscribing was addressed 👍

**Pull request merged**